### PR TITLE
docs: update conventions for branch names, commit messages and coding styles

### PR DESCRIPTION
**Pull request by leeping-ng**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -66,43 +67,89 @@ Thank you for your contributions!!
...
+`<type>` must be one of the options described in the above [Branch Names](#branch-names) section.
+
+### Code Styles
+
+We follow the PEP8 style guide, with PEP484 type hinting for functions and methods, and use [Black](https://black.readthedocs.io/en/stable/) code formatter to ensure consistent code format.
+
+The following are commonly used conventions which we adhere to:
+- Imports
+  - Absolute imports instead of relative imports (e.g. avoid `from ...nodes.node import AbstractNode`)
+  - Use [isort](https://github.com/PyCQA/isort) to order imports
```
Comment (liyier90) [inline CONTRIBUTING.md:123]: `isort` requires some additional settings to play nice with `black` and to recognize `peekingduck` as first-party import.

The following section
```
[tool.isort]
profile = "black"
```
needs to be added to `pyproject.toml`.

Or this point can be removed if we do not play on enforcing using `isort` to sort imports.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -66,43 +67,89 @@ Thank you for your contributions!!
...
-  │       │             └─⫸ Summary in present tense. Not capitalized. No period at the end.
-  │       │
-  │       └─⫸ Commit Scope: core|cli|changelog|packaging|models|heuristics|cicd-infra|docs-infra|community
+<type>: <short summary>
+  │            │
+  │            └─⫸ Summary in present tense. Not capitalized. No period at the end.
   │
-  │
-  └─⫸ Commit Type: build|ci|docs|feat|fix|perf|refactor|test
+  └─⫸ Commit Type: cicd|docs|feat|fix|node|refactor|test
```
Comment (liyier90) [inline CONTRIBUTING.md:112]: Which <type> to use for `requirements.txt` and `setup.cfg` related changes?

File: CONTRIBUTING.md (hunk 1 above)
Comment (leeping-ng) [reply CONTRIBUTING.md:123]: Let's just remove this point then

File: CONTRIBUTING.md (hunk 2 above)
Comment (leeping-ng) [reply CONTRIBUTING.md]: Added new type: 
```build: Updates to dependencies and package building```

**Review by liyier90 — APPROVED** [review]
LGTM

**Comment by ongtw** [comment]
Suggestions for Code Style changes (after re-reading with a clear mind):

- Strings
  - f-strings can also be used for string concatenation. 

- Naming
  - ~~`CapWords` for `ClassName`, `ExceptionName`~~
    `PascalCase` for `ClassName`, `ExceptionName`
  - ~~Fully capitalised for `GLOBAL_CONSTANT_NAME`~~
    `snake_case` all caps for `GLOBAL_CONSTANT_NAME`, or
    `SNAKE_CASE_ALL_CAPS` for `GLOBAL_CONSTANT_NAME` 
  - ~~`_name` postfix for file names, e.g. `output.mp4`~~
      `_file` postfix for file names, e.g. `media.mp4`
    In this case, variables `media_file` and `media_name` would be clearer: the former refers to the file `media.mp4` while the latter refers to the name of the media (e.g. "Video recording 25 Dec 2021").

**Pull request merged**